docker/Dockerfile.musl

```diff
@@ -0,0 +1,46 @@
+FROM alpine:3.11
```

Comment:
I see that Alpine 3.11 uses musl libc 1.1.24, released in October. Assuming musl adds symbols between releases, anyone with older versions won't be able to link against the binary. That's why the other Dockerfile uses an old Debian and compiles from source. So I think that might be one pre-req to merging this.

Reply:
Got it. I can make the change to an older version of alpine. How far back are you looking to go though?

Reply:
@chearon I was looking at this again and I think it makes sense to leave musl at 1.1.24.  They are [planning to EOL](https://musl.libc.org/releases.html) the 1.1 branch with the release of 1.2 in February but will be back-porting bug fixes down to 1.1.25 in the future. Also, all versions prior to 1.1.24 have security issues. 

Similar to how musl will only be supported for node versions >= 12 (released in April 2019) I don't think it's unreasonable to only "officially" support musl >= 1.1.24 (released in Oct 2019).  What do you think?

Reply:
Sorry to jump in - given the EOL, eventual complications and that there are no better options either ways so far - having recent support now is better than having legacy support but too late.
Probably a lot of people in the K8S world are using Alpine as a base and are struggling / consuming power to build these for nothing :)